industry_benchmarks/utils/data_gathering.py

```diff
@@ -453,14 +474,267 @@ def gather_ligand_scores(
+    -------
+        results: dict[str, tuple[unit.Quantity, unit.Quantity]]
+        The extracted DDG and uncertainty for each transformation repeat
+        > {"solvent_liganda_ligandb_repeat_0": (ddg, uncertainty) ...
+
+    """
+    # workout the expected number of results
+    # assuming 3 repeats of each solvent and complex transformation
+    missing_results = []
+    all_results = {}
+    for edge in alchemical_network.edges:
+        all_results[edge.name] = []
+    expected_results = len(all_results) * 3
+
+    # map the transformation to the results files
+    for results_folder in results_folders:
+        for results_file in tqdm.tqdm(results_folder.glob("*.json"), desc=f"Processing results in {results_folder}", ncols=80):
+            # run checks on the end results
+            result = load_results_file(file_name=results_file)
+            if result is not None:
+                # get the estimate which will be saved
+                ddg, uncertainty = get_estimate(result)
+                # find the cleaned up results file
+                simulation_data_file = find_data_folder(result=result)
+                # work out the name of the transform and add it to the dict
+                transformation_name = get_transform_name(result=result, alchemical_network=alchemical_network)
+                # collect the paths to the results files and the structural data
+                if transformation_name in all_results and simulation_data_file is not None:
+                    all_results[transformation_name].append((ddg, uncertainty, simulation_data_file))
+                elif transformation_name not in all_results:
+                    error_message = (f"Found a result for {transformation_name} which was not expected "
+                                     f"from the alchemical network.")
+                    raise ValueError(error_message)
+
+    # Write stats on the number of transformations found
+    found_results = sum([len(v) for v in all_results.values()])
+    print(f"Total results found {found_results}/{expected_results} indicating {expected_results - found_results} failed transformations.")
+    # check we have a connected network
+    if not check_network_is_connected(results_data=all_results, alchemical_network=alchemical_network):
+        raise ValueError("The network built from the complete results is disconnected, some simulations may still be"
+                         "running or needed restarting. Reproducible edge failures may require extra edges which "
+                         "can be generated using the `fix_networks.py` script.")
+
+    estimates = {}
+    # move the results to the output folder and collect the estimates
+    for transformation_name, results in tqdm.tqdm(all_results.items(), desc="Collecting files", total=len(all_results), ncols=80):
+        for i, ddg, uncertainty, results_dir in enumerate(results):
+            # store the per repeat estimates
+            repeat = f"repeat_{i}"
```

Comment:
I wonder if we want to store the result with the actual repeat number. Would this be helpful in case we have to do some troubleshooting later? I think it won't be necessary for our analysis, just in case we have to ask partners if they could look into a specific repeat it may be helpful? I'm not sure if this is necessary..what do you think?

Reply:
We definitely should be storing our results with the repeat numbers, i.e.

ligand A:
  complex repeat 1:
    results:
  complex repeat 2:
    results:
etc...